### ISSUE TITLE
Apple Mach-O Linker Warning Introduced 

### ISSUE THREAD

**Issue by axandreo** [Quick/Nimble#540]
### What did you do?

ran pod update to update my list of pods; Nimble is a part of that list. Nimble updated from 7.1.1 to 7.1.2.

This also changed the source? `(was 7.1.1 and source changed to "https://github.com/CocoaPods/Specs.git" from "https://github.com/cocoapods/specs.git")`

### What did you expect to happen?

I expected not to see this warning: `Linking against a dylib which is not safe for use in application extensions: /Applications/Xcode.app/path/to/libSwiftXCTest.dylib`

### What actually happened instead?

The warning I expected not to see, was there.

### Environment

List the software versions you're using:
Latest Quick Version: 1.3.0
Latest Nimble Version: 7.1.2
Latest Xcode Version: 9.4
Latest Swift Version: Swift 4.x
Cocoapods version 1.5.3

### Project that demonstrates the issue

An example project can be found at this repository: axandreo/AppleMachONimbleIssueProject
To get the warning, build the project for testing: Product > Build For > Testing or Shift, Command, U (I believe that is the keyboard shortcut).

NOTE:
While creating this project cocoapods also gave me these warnings:
```
[!] The `NimbleMachOIssueTests [Debug]` target overrides the `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` build setting defined in `Pods/Target Support Files/Pods-NimbleMachOIssueTests/Pods-NimbleMachOIssueTests.debug.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.

[!] The `NimbleMachOIssueTests [Release]` target overrides the `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` build setting defined in `Pods/Target Support Files/Pods-NimbleMachOIssueTests/Pods-NimbleMachOIssueTests.release.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.
```
I didn't expect these either, has that always been an issue?

**Comment by ikesyo** [comment]
Since that is just an warning and the library is only for testing, that is safely ignored. But looks like we can mitigate the situation with specifying `-Xlinker -no_application_extension` to `OTHER_LDFLAGS`.

**Comment by ikesyo** [comment]
Addressed by #541.
